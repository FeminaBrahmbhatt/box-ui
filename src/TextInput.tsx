import React, { FC } from "react";
import { TextInputProps } from "./@interfaces";
import { Typography } from "./Typography";
import classNames from "classnames";
import { useIconClassName } from "./hooks/useIconClassName";

export const TextInput: FC<TextInputProps> = ({
  type,
  value,
  label,
  placeholder,
  error,
  helperText,
  disabled,
  LeadingIcon,
  TrailingIcon,
  leadingText,
  handleChange,
}) => {
  const { setClassName } = useIconClassName();

  return (
    <div>
      {label && (
        <Typography
          variant="sm"
          customWeight="medium"
          customColor="text-gray-700 dark:text-white"
          className="mb-1.5"
        >
          {label}
        </Typography>
      )}
      <div
        className={classNames("relative", { "flex items-center": leadingText })}
      >
        <div className="h-11 absolute shadow-sm rounded-lg w-full flex justify-between items-center px-3.5 pointer-events-none">
          {LeadingIcon ? setClassName(LeadingIcon, "text-gray-500") : <div />}
          {TrailingIcon &&
            setClassName(
              TrailingIcon,
              classNames({ "text-error-500": error, "text-gray-400": !error })
            )}
        </div>
        {leadingText && (
          <Typography
            variant="sm"
            customWeight="regular"
            className={classNames(
                "flex items-center h-11 text-lg text-gray-500 pl-3.5 pr-3 border border-r-0 rounded-l-lg border-gray-300 dark:border-gray-500",
                {
                  "bg-gray-50 dark:bg-gray-700": disabled,
                  "dark:bg-gray-800": !disabled,
                },
              )}
          >
            {leadingText}
          </Typography>
        )}
        <input
          type={type}
          value={value}
          placeholder={placeholder}
          onChange={handleChange}
          disabled={disabled}
          className={classNames(
            "w-full select-none text-gray-900 dark:text-white text-md border h-11 px-2",
            {
              "pl-9": LeadingIcon,
              "pr-9": TrailingIcon,
              "rounded-l-0 rounded-r-lg": leadingText,
              "rounded-lg": !leadingText,
              "border-gray-300 dark:border-gray-500 focus:ring-4 focus:border-primary-300 dark:focus:border-gray-100 focus:ring-primary-100 dark:focus:ring-gray-100 dark:focus:ring-opacity-20":
                !error,
              "border-error-300 focus:ring-4 focus:border-error-300 focus:ring-error-100":
                error,
              "bg-white dark:bg-gray-800": !disabled,
              "bg-gray-50 dark:bg-gray-700": disabled,
            },
          )}
        />
      </div>
      {error && (
        <Typography
          variant="sm"
          customWeight="regular"
          className="mt-1.5 text-error-500"
        >
          {error}
        </Typography>
      )}
      {helperText && (
        <Typography
          variant="sm"
          customWeight="regular"
          className="mt-1.5 text-gray-500"
        >
          {helperText}
        </Typography>
      )}
    </div>
  );
};
