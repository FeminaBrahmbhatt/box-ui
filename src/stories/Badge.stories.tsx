import React from 'react';
import {BadgeProps} from '../@interfaces';
import { Badge } from '../Badge';
import {images} from '../data/images'
import {Meta, StoryObj} from '@storybook/react';
import { StoryLayout } from '../StoryLayout';
import { FiArrowRight, FiStar } from 'react-icons/fi';
import { IconWrapper } from '../utils/IconWrapper';

export default {
    title: 'Example/Badge',
    component: Badge,
    parameters: {
        layout: 'centered',
    },
    tags: ['autodocs'],
} as Meta;
interface Props extends BadgeProps {
    darkMode: boolean;
}

export const Default:StoryObj<Props> = {
    args: {
        darkMode: false,
        variant: "success",
        size: "md",
        children:<span>Label</span>
    }
}

export const LeadingIconBadge:StoryObj<Props>  = {
    args: {
        darkMode: false,
        variant: "success",
        size: "md",
        LeadingIcon: <IconWrapper Icon={FiStar} data-testid="leadingIcon"/>
    },
    render: (args) => {
        return (
            <StoryLayout {...args} className="inline-flex flex-col space-y-2">
                <Badge {...args}>
                    Label
                </Badge>
            </StoryLayout>
        )
    }
}

export const ImageIconBadge:StoryObj<Props>  = {
    args: {
        darkMode: false,
        variant: "gray",
        size: "md",
        LeadingIcon: <img src={images.NL} alt="nl" className="w-4 h-4" data-testid="iconOnly" />
    },
    render: (args) => {
        return (
            <StoryLayout {...args} className="inline-flex flex-col space-y-2">
                <Badge {...args}>
                    Newzeland
                </Badge>
            </StoryLayout>
        )
    }
}

export const TrailingIconBadge:StoryObj<Props>  = {
    args: {
        darkMode: false,
        variant: "primary",
        size: "md",
        TrailingIcon: <IconWrapper Icon={FiArrowRight} data-testid="trailingIcon"/>
    },
    render: (args) => {
        return (
            <StoryLayout {...args} className="inline-flex flex-col space-y-2">
                <Badge {...args}>
                    Label
                </Badge>
            </StoryLayout>
        )
    }
}

