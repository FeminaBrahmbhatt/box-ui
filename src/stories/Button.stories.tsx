import React from "react";
import type { Meta, StoryObj } from "@storybook/react";
import { Button } from "../Button";
import { ButtonProps } from "../@interfaces";
import { FiArrowLeft, FiArrowRight, FiStar } from "react-icons/fi";
import { StoryLayout } from "../StoryLayout";
import { IconWrapper } from "../utils/IconWrapper";

export default{
  title: "Example/Button",
  component: Button,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
} as Meta;
interface Props extends ButtonProps {
  darkMode: boolean;
}

type Story = StoryObj<Props>;

export const Default: Story = {
  args: {
    darkMode: false,
    children: "Button",
    size: 'md',
    variant: 'primary',
    disabled: false,
    LeadingIcon: undefined,
    TrailingIcon: undefined,
    IconOnly: undefined,
    className: ''
  },
  render: (args) => {
    return (
      <StoryLayout {...args} className="flex inline-flex flex-row space-x-2">
        <Button {...args}>Primary</Button>
      </StoryLayout>
    );
  },
};

export const Variants: Story = {
  args: {
    darkMode: false,
    variant: "primary",
  },
  render: (args) => {
    return (
      <StoryLayout {...args} className="flex inline-flex flex-row space-x-2">
        <Button variant="primary">Primary</Button>
        <Button variant="secondary">Secondary</Button>
        <Button variant="secondaryGray">Secondary Gray</Button>
        <Button variant="tertiary">Tertiary</Button>
        <Button variant="tertiaryGray">Tertiary Gray</Button>
      </StoryLayout>
    );
  },
};

export const Sizes: Story = {
  args: {
    darkMode: false,
    variant: "primary",
    size: "md",
  },
  render: (args) => {
    return (
      <StoryLayout {...args} className="flex inline-flex flex-row space-x-2">
        <Button size="sm">sm</Button>
        <Button size="md">md</Button>
        <Button size="lg">lg</Button>
        <Button size="xl">xl</Button>
        <Button size="2xl">2xl</Button>
      </StoryLayout>
    );
  },
};

export const TrailingIconButton: Story = {
  args: {
    darkMode: false,
    children: "Button",
    TrailingIcon: <IconWrapper Icon={FiArrowRight} data-testid="trailingIcon"/>,
  },  
  render: (args) => (
    <StoryLayout {...args}>
      <Button {...args} />
    </StoryLayout>
  ),
};

export const LeadingIconButton: Story = {
  args: {
    darkMode: false,
    children: "Button",
    LeadingIcon: <IconWrapper Icon={FiArrowLeft} data-testid="LeadingIcon"/>,
  },
  render: (args) => (
    <StoryLayout {...args}>
      <Button {...args} />
    </StoryLayout>
  ),
};

export const IconOnlyButton: Story = {
  args: {
    darkMode: false,
    IconOnly: <IconWrapper Icon={FiStar} data-testid="iconOnly"/>,
  },
  render: (args) => (
    <StoryLayout {...args}>
      <Button {...args} />
    </StoryLayout>
  ),
};
