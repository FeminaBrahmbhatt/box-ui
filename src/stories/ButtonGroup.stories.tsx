import React, { useState } from "react";
import { Meta, StoryObj } from "@storybook/react";
import { ButtonGroupProps, ButtonGroup } from "../ButtonGroup";
import { StoryLayout } from "../StoryLayout";
import { options1, options2, options3 } from "../data/options";

export default {
  title: "Example/ButtonGroup",
  component: ButtonGroup,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} as Meta;
interface Props extends ButtonGroupProps {
  darkMode: boolean;
}

const LabelButtonGroupComponent = (args: Props) => {
  const [activeItem1, setActiveItem1] = useState<string>(options1[1].value);

  return (
    <StoryLayout {...args}>
      <ButtonGroup
        {...args}
        options={options1}
        activeOption={activeItem1}
        setActiveOption={setActiveItem1}
      />
    </StoryLayout>
  );
};

export const LabelButtonGroup: StoryObj<Props> = {
  args: {
    darkMode: false,
  },
  render: (args) =>  <LabelButtonGroupComponent {...args} />,
};

const IconButtonGroupComponent = (args: Props) => {
  const [activeView, setViewOption] = useState<string>("list");

  return (
    <StoryLayout {...args}>
      <ButtonGroup
        {...args}
        options={options3}
        activeOption={activeView}
        setActiveOption={setViewOption}
      />
    </StoryLayout>
  );
};

export const IconButtonGroup: StoryObj<Props> = {
  args: {
    darkMode: false,
  },
  render: (args) => <IconButtonGroupComponent {...args} />,
};

const LabelIconButtonGroupComponent = (args: Props) => {
  const [activeItem2, setActiveItem2] = useState<string>(options2[1].value);

  return (
    <StoryLayout {...args}>
      <ButtonGroup
        {...args}
        options={options2}
        activeOption={activeItem2}
        setActiveOption={setActiveItem2}
      />
    </StoryLayout>
  );
};

export const LabelIconButtonGroup: StoryObj<Props> = {
  args: {
    darkMode: false,
  },
  render: (args) => <LabelIconButtonGroupComponent {...args} />,
};
