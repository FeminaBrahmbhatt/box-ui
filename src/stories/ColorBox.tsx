import React from "react";
import { IColor } from "./@interfaces";

export const ColorBox = ({ code, hex, bgClass }: IColor) => (
<div
    key={bgClass}
    className="flex flex-col rounded-lg shadow-lg w-26 h-39 dark:bg-white">
      <div className={`h-20 ${bgClass} rounded-t-lg`} />
      <div className="mx-3 my-3">
          {code}
          <br/>
          {hex}
      </div>
    </div>
);