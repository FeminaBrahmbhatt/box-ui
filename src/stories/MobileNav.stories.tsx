import React, { useState } from "react";
import { Meta, StoryObj } from "@storybook/react";
import { MobileNavProps } from "../@interfaces";
import { SideNav } from "../Nav/SideNav";
import { StoryLayout } from "../StoryLayout";
import { navItemsTop, navItemsBottom } from "../data/navItems";
import { MobileNavbar } from "../Nav/MobileNav";
import { FiX } from "react-icons/fi"; // Directly import FiX
import type { IconBaseProps } from "react-icons";

const CloseIcon = FiX as React.FC<IconBaseProps>;
// Wrapper to fix TS JSX issues
 
export default {
  title: "Example/MobileNav",
  component: MobileNavbar,
} as Meta;

interface Props extends MobileNavProps {
  darkMode: boolean;
}

const MobileNavWithState = (args: Props) => {
  const [open, setOpen] = useState<boolean>(false);

  const handleToggle = () => {
    setOpen(!open);
  };

  return (
    <StoryLayout {...args}>
      <MobileNavbar open={open} toggleOpen={handleToggle} />

      {open && (
        <div
          data-testid="backdrop"
          className="fixed top-0 left-0 z-40 w-screen h-screen bg-gray-500 cursor-pointer bg-opacity-80"
          onClick={() => setOpen(false)}
        >
           <CloseIcon
            data-testid="close-icon"
            size={40}
            className="fixed text-white top-5 right-9"
            role="button"
            aria-label="Close sidebar"
          />
        </div>
      )}

      {open && (
        <SideNav
          className="relative z-50 h-screen -mt-20"
          navItemsTop={navItemsTop}
          navItemsBottom={navItemsBottom}
          open={true}
          setOpen={handleToggle}
          username="Veronica Woods"
          email="veronica@example.com"
        />
      )}
    </StoryLayout>
  );
};

export const Default: StoryObj<Props> = {
  args: {
    darkMode: false,
  },
  render: (args) => <MobileNavWithState {...args} />,
};
