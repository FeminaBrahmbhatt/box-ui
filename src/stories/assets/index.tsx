
export type CountryISO = "BE" | "DE" | "FR" | "IT" | "NL" | "RU" | "US";

type image = CountryISO;

// Development images
const devImages: Record<string, image> = {
  NL: require("./nl.svg")
};

// Production images (CDN)
// const prodImages: Record<string, image> = {
//   NL: "https://res.cloudinary.com/tailwindcss/image/upload/v1635279277/nl_tpy2ab.svg",
// };

export const assets = devImages;