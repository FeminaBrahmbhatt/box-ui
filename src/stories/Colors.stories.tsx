import React from "react";
import { Meta, StoryObj } from "@storybook/react";
import { ColorBox } from "../ColorBox";
import { colors } from "../data/colors";
import { StoryLayout } from "../StoryLayout";

export default {
  title: "Colors",
  parameters: {
    controls: { expanded: true },
  },
} as Meta;

interface Props {
  darkMode: boolean;
}

export const Default: StoryObj<Props> = {
  args: {
    darkMode: false,
  },
  render: (args) => (
    <StoryLayout {...args} className="grid grid-cols-2 gap-4 md:grid-cols-4 lg:grid-cols-6 xl:grid-cols-11">
      {colors.map((color) => (
        <ColorBox key={color.bgClass} code={color.code} hex={color.hex} bgClass={color.bgClass}/>
      ))}
    </StoryLayout>
  ),
};
