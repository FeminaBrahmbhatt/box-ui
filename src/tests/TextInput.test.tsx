import React from 'react';
import '@testing-library/jest-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import { composeStories } from '@storybook/react';
import * as stories from '../stories/TextInput.stories';

const {Default, LeadingTextInput} = composeStories(stories);

describe('ðŸ§ª TextInput Story Tests (Direct function call style)', () => {
    test('Default story renders with email input and error', () => {
      const Story = Default;
      const { container } = render(
        Story(Story.args ?? {})
      );
  
      expect(screen.getAllByLabelText(/Email/i)[0]).toBeInTheDocument();
      expect(screen.getByText(/This is an error message/i)).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });
  
    test('LeadingTextInput story renders with https prefix and error', () => {
      const Story = LeadingTextInput;
      const { container } = render(
        Story(Story.args ?? {})
      );
  
      expect(screen.getAllByLabelText(/Website/i)[0]).toBeInTheDocument();
      expect(screen.getAllByText('https://')[0]).toBeInTheDocument();
      expect(screen.getByText(/This is an error message/i)).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });
  
    test('typing works in Default TextInput', () => {
      const Story = Default;
      render(Story(Story.args ?? {}));
  
      const input = screen.getAllByLabelText(/Email/i)[0] as HTMLInputElement;
  
      fireEvent.change(input, { target: { value: 'femina@example.com' } });
      expect(input.value).toBe('femina@example.com');
    });
  });