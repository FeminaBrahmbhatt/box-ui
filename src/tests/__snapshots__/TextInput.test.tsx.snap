// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª TextInput Story Tests (Direct function call style) Default story renders with email input and error 1`] = `
<div>
  <div
    class="-m-4"
  >
    <div
      class="p-4"
    >
      <div>
        <label
          for="textinput-email"
        >
          <p
            class="text-sm font-medium mb-1.5 text-gray-700 dark:text-white"
          >
            Email
          </p>
        </label>
        <div
          class="relative"
        >
          <div
            class="h-11 absolute shadow-sm rounded-lg w-full flex justify-between items-center px-3.5 pointer-events-none"
          >
            <svg
              class="text-gray-500"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
              />
              <polyline
                points="22,6 12,13 2,6"
              />
            </svg>
            <svg
              class="text-gray-400"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <path
                d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
              />
              <line
                x1="12"
                x2="12.01"
                y1="17"
                y2="17"
              />
            </svg>
          </div>
          <input
            class="w-full select-none text-gray-900 dark:text-white text-md border h-11 px-2 pl-9 pr-9 rounded-lg border-gray-300 dark:border-gray-500 focus:ring-4 focus:border-primary-300 dark:focus:border-gray-100 focus:ring-primary-100 dark:focus:ring-gray-100 dark:focus:ring-opacity-20 bg-white dark:bg-gray-800"
            id="textinput-email"
            placeholder="veronica@example.com"
            type="email"
            value=""
          />
        </div>
        <label
          class="text-sm font-normal mt-1.5 text-gray-500"
        >
          This is a hint text to help the user.
        </label>
      </div>
      <div
        class="mb-4"
      />
      <div>
        <label
          for="textinput-email"
        >
          <p
            class="text-sm font-medium mb-1.5 text-gray-700 dark:text-white"
          >
            Email
          </p>
        </label>
        <div
          class="relative"
        >
          <div
            class="h-11 absolute shadow-sm rounded-lg w-full flex justify-between items-center px-3.5 pointer-events-none"
          >
            <svg
              class="text-gray-500"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
              />
              <polyline
                points="22,6 12,13 2,6"
              />
            </svg>
            <svg
              class="text-error-500"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <path
                d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
              />
              <line
                x1="12"
                x2="12.01"
                y1="17"
                y2="17"
              />
            </svg>
          </div>
          <input
            class="w-full select-none text-gray-900 dark:text-white text-md border h-11 px-2 pl-9 pr-9 rounded-lg border-error-300 focus:ring-4 focus:border-error-300 focus:ring-error-100 bg-white dark:bg-gray-800"
            id="textinput-email"
            placeholder="veronica@example.com"
            type="email"
            value=""
          />
        </div>
        <label
          class="text-sm font-normal mt-1.5 text-error-500"
        >
          This is an error message.
        </label>
        <label
          class="text-sm font-normal mt-1.5 text-gray-500"
        >
          This is a hint text to help the user.
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`ðŸ§ª TextInput Story Tests (Direct function call style) LeadingTextInput story renders with https prefix and error 1`] = `
<div>
  <div
    class="-m-4"
  >
    <div
      class="p-4"
    >
      <div>
        <label
          for="textinput-website"
        >
          <p
            class="text-sm font-medium mb-1.5 text-gray-700 dark:text-white"
          >
            Website
          </p>
        </label>
        <div
          class="relative flex items-center"
        >
          <div
            class="h-11 absolute shadow-sm rounded-lg w-full flex justify-between items-center px-3.5 pointer-events-none"
          >
            <div />
            <svg
              class="text-gray-400"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <path
                d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
              />
              <line
                x1="12"
                x2="12.01"
                y1="17"
                y2="17"
              />
            </svg>
          </div>
          <p
            class="text-sm font-normal flex items-center h-11 text-lg text-gray-500 pl-3.5 pr-3 border border-r-0 rounded-l-lg border-gray-300 dark:border-gray-500 dark:bg-gray-800 text-black dark:text-white"
          >
            https://
          </p>
          <input
            class="w-full select-none text-gray-900 dark:text-white text-md border h-11 px-2 pr-9 rounded-l-0 rounded-r-lg border-gray-300 dark:border-gray-500 focus:ring-4 focus:border-primary-300 dark:focus:border-gray-100 focus:ring-primary-100 dark:focus:ring-gray-100 dark:focus:ring-opacity-20 bg-white dark:bg-gray-800"
            id="textinput-website"
            placeholder="example.com"
            type="text"
            value=""
          />
        </div>
        <label
          class="text-sm font-normal mt-1.5 text-gray-500"
        >
          This is a hint text to help the user.
        </label>
      </div>
      <div
        class="mb-4"
      />
      <div>
        <label
          for="textinput-website"
        >
          <p
            class="text-sm font-medium mb-1.5 text-gray-700 dark:text-white"
          >
            Website
          </p>
        </label>
        <div
          class="relative flex items-center"
        >
          <div
            class="h-11 absolute shadow-sm rounded-lg w-full flex justify-between items-center px-3.5 pointer-events-none"
          >
            <div />
            <svg
              class="text-error-500"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <line
                x1="12"
                x2="12"
                y1="8"
                y2="12"
              />
              <line
                x1="12"
                x2="12.01"
                y1="16"
                y2="16"
              />
            </svg>
          </div>
          <p
            class="text-sm font-normal flex items-center h-11 text-lg text-gray-500 pl-3.5 pr-3 border border-r-0 rounded-l-lg border-gray-300 dark:border-gray-500 dark:bg-gray-800 text-black dark:text-white"
          >
            https://
          </p>
          <input
            class="w-full select-none text-gray-900 dark:text-white text-md border h-11 px-2 pr-9 rounded-l-0 rounded-r-lg border-error-300 focus:ring-4 focus:border-error-300 focus:ring-error-100 bg-white dark:bg-gray-800"
            id="textinput-website"
            placeholder="example.com"
            type="text"
            value=""
          />
        </div>
        <label
          class="text-sm font-normal mt-1.5 text-error-500"
        >
          This is an error message.
        </label>
      </div>
    </div>
  </div>
</div>
`;
